homeassistant/components/mqtt/humidifier.py

```diff
@@ -0,0 +1,456 @@
+            "HUMIDITY_RESET": config[CONF_PAYLOAD_RESET_HUMIDITY],
+            "MODE_RESET": config[CONF_PAYLOAD_RESET_MODE],
+        }
+        if CONF_MODE_COMMAND_TOPIC in config and CONF_AVAILABLE_MODES_LIST in config:
+            self._available_modes = config[CONF_AVAILABLE_MODES_LIST]
+        else:
+            self._available_modes = []
+        if self._available_modes:
+            self._attr_supported_features = SUPPORT_MODES
+        else:
+            self._attr_supported_features = 0
+
+        optimistic = config[CONF_OPTIMISTIC]
+        self._optimistic = optimistic or self._topic[CONF_STATE_TOPIC] is None
+        self._optimistic_target_humidity = (
+            optimistic or self._topic[CONF_TARGET_HUMIDITY_STATE_TOPIC] is None
+        )
+        self._optimistic_mode = optimistic or self._topic[CONF_MODE_STATE_TOPIC] is None
+
+        for tpl_dict in [self._command_templates, self._value_templates]:
+            for key, tpl in tpl_dict.items():
+                if tpl is None:
+                    tpl_dict[key] = lambda value: value
+                else:
+                    tpl.hass = self.hass
+                    tpl_dict[key] = tpl.async_render_with_possible_json_value
+
+    async def _subscribe_topics(self):
+        """(Re)Subscribe to topics."""
+        topics = {}
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def state_received(msg):
+            """Handle new received MQTT message."""
+            payload = self._value_templates[CONF_STATE](msg.payload)
+            if not payload:
+                _LOGGER.debug("Ignoring empty state from '%s'", msg.topic)
+                return
+            if payload == self._payload["STATE_ON"]:
+                self._state = True
+            elif payload == self._payload["STATE_OFF"]:
+                self._state = False
+            self.async_write_ha_state()
+
+        if self._topic[CONF_STATE_TOPIC] is not None:
+            topics[CONF_STATE_TOPIC] = {
+                "topic": self._topic[CONF_STATE_TOPIC],
+                "msg_callback": state_received,
+                "qos": self._config[CONF_QOS],
+            }
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def target_humidity_received(msg):
+            """Handle new received MQTT message for the taret humidity."""
+            rendered_target_humidity_payload = self._value_templates[ATTR_HUMIDITY](
+                msg.payload
+            )
+            if not rendered_target_humidity_payload:
+                _LOGGER.debug("Ignoring empty target humidity from '%s'", msg.topic)
+                return
+            if rendered_target_humidity_payload == self._payload["HUMIDITY_RESET"]:
+                self._target_humidity = None
+                self.async_write_ha_state()
+                return
+            try:
+                target_humidity = int(float(rendered_target_humidity_payload))
```

Comment:
Why first cast to a float and then truncate to an int? Also, don't we want to round instead?

Reply:
If the value template gives us a float value, it passes it as a string, E.g. '60.0'. This cannot be converted to int directly using int() or round(). So to make sure it is a number, I convert it to float first. To your suggestion I have replaced int() with round, which is a better fit.